### PR TITLE
feat: support kda fla_npu operator build and execute_linear integration.

### PR DESCRIPTION
**Pull request by sunbaosong**
## 概述

为 NPU 后端接入 KDA(Kimi Delta Attention)的 `execute_linear`:conv1d 保持纯 torch,delta-rule 调用 `fla_npu` 融合算子(`chunk_kda_fwd` for prefill,`recurrent_kda` for decode,含 MTP 多 token-per-seq varlen)。重新对齐 upstream 的 `LayerCache` dataclass,conv/ssm 状态读自框架池。

> `fla_npu` 是外部依赖,不在此 PR 编译安装;运行 KDA 层前需自行安装 fla_npu(暴露 `chunk_kda_fwd` / `recurrent_kda`)。

### 1. `execute_linear` 接入

- **`xllm/python/attention/backend.py`**:新增 `AttentionBackend.execute_linear` 抽象入口(conv1d + delta-rule,操作框架 conv/ssm 状态),契约与 `execute_mla` 对称。
- **`xllm/python/attention/npu_paged_attention.py`**:
  - `NpuPagedAttentionBackend.execute_linear`:多并发走 **batched varlen**(引擎 flatten 多序列为 `[1,T,H,D]`,delta-rule 用 `cu_seqlens` **单次调用** fla_npu,非逐序列循环);单序列统一走 varlen(`cu_seqlens=[0,S]` 特例)。
  - `_kda_fla_npu_varlen`:decode→`recurrent_kda`(varlen),prefill→`chunk_kda_fwd`(varlen),无纯 torch fallback。
  - conv/ssm 直读 `_kv_caches`(单一真相源,无 projection list);状态契约用 assert 钉死(`linear_state_indices`/`conv`/`ssm`/`has_initial_state` 必须已设,引擎预分配池并每序列一槽)。

### 2. conv1d 辅助函数

- **`xllm/python/kernels_npu/kda.py`**(新增):纯 torch conv1d 辅助函数 —— `_l2norm`、`_causal_conv1d_fn`、`_causal_conv1d_update`。无模型耦合,任何环境均可安全 import。

### 单算子对等(fla_npu vs 纯 torch 参考)

`fla_npu` 算子 vs 纯 torch 参考(`chunk_kimi_delta_attention`/`recurrent_kimi_delta_attention`),同一组输入,NPU 容器实跑,`B=1 / H=16 / K=V=128`,prefill `S=64`、decode `S=1`。

| 路径 | out maxdiff | state maxdiff |
|------|------------|---------------|
| prefill(`chunk_kda_fwd` vs `chunk_kimi`) | 3.9e-3(~0.4%) | 2.65e-2(~0.07%) |
| decode(`recurrent_kda` vs `recurrent_kimi`) | 6.1e-5 | 4.8e-7 |

> 相对量级:prefill out absmax 0.918、state absmax 37;decode 近乎精确。均在 bf16 精度内。

### CODE REVIEW

File: xllm/compiler/fla_npu/fla_npu_install.py [hunk 1]
```diff
@@ -0,0 +1,213 @@
+"""Build and install the fla_npu AscendC fused KDA ops.
```
Comment (zhang-minchao) [inline xllm/compiler/fla_npu/fla_npu_install.py]: 安装fla的代码，不应该放在这个目录吧

File: xllm/python/attention/npu_paged_attention.py [hunk 2]
```diff
@@ -122,6 +123,155 @@ def is_mla(self) -> bool:
...
+        qkv_dim = layer.qkv_dim
+        is_prefill = metadata.is_prefill or metadata.is_chunked_prefill
+
+        idx = metadata.linear_state_indices
+        assert idx is not None, "execute_linear requires metadata.linear_state_indices"
+        num_seqs = idx.shape[0]
+        if idx.dtype != torch.int64:
+            idx = idx.to(torch.int64)
+        conv_state = conv_cache.index_select(0, idx)
+        conv_state = conv_state.transpose(1, 2).contiguous()
```
Comment (zhang-minchao) [inline xllm/python/attention/npu_paged_attention.py:158]: 每次都要临时transpose吗？

File: xllm/python/attention/npu_paged_attention.py [hunk 3]
```diff
@@ -124,6 +123,158 @@ def is_mla(self) -> bool:
...
+            cs = conv_state[s:s + 1]       # [1, state_len, conv_dim]
+            seg_len = t1 - t0
+            if is_prefill:
+                cin = torch.cat([cs, seg], dim=1)
+                out = _causal_conv1d_fn(
+                    cin, conv_weight, activation
+                )[:, -seg_len:, :]
+                cs_new = cin[:, -conv_state_len:, :]
+                if cs_new.shape[1] < conv_state_len:
+                    cs_new = F.pad(
```
Comment (yingxudeng) [inline xllm/python/attention/npu_paged_attention.py]: 这个分支真的会走到吗？还是死代码

File: xllm/python/kernels_npu/kda.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+from __future__ import annotations
+
+import torch
+import torch.nn.functional as F
+
+
+def _l2norm(x: torch.Tensor, dim: int = -1, eps: float = 1e-6) -> torch.Tensor:
+    """FLA-style l2norm: sqrt(sum(x^2)+eps) then divide (NOT F.normalize)."""
+    inv_norm = torch.sqrt((x * x).sum(dim=dim, keepdim=True) + eps)
```
Comment (yingxudeng) [inline xllm/python/kernels_npu/kda.py]: torch.sqrt((x * x).sum(dim=dim, keepdim=True) + eps) 这样算出来的不是 norm 本身吗？为啥叫 inv_norm

File: xllm/compiler/fla_npu/fla_npu_install.py (hunk 1 above)
Comment (sunbaosong) [reply xllm/compiler/fla_npu/fla_npu_install.py]: 已删除编译过程

File: xllm/python/attention/npu_paged_attention.py (hunk 3 above)
Comment (sunbaosong) [reply xllm/python/attention/npu_paged_attention.py]: 已删除

File: xllm/python/kernels_npu/kda.py (hunk 4 above)
Comment (sunbaosong) [reply xllm/python/kernels_npu/kda.py]: 确实，已改正
